functional_tests/scylla_operator/test_functional.py

```diff
@@ -564,6 +562,27 @@ def test_deploy_helm_with_default_values(db_cluster: ScyllaPodCluster):
         db_cluster.k8s_cluster.kubectl(f"delete namespace {namespace}")
 
 
+@pytest.mark.restart_is_used
+def test_rolling_config_change_internode_compression(db_cluster, scylla_yaml):
+    """
+    Cover logic of disrupt_rolling_config_change_internode_compression nemesis
+    """
+    internode_compression_option_name = "internode_compression"
+
+    with db_cluster.nodes[0].actual_scylla_yaml() as props:
+        original_compression = dict(props).get(internode_compression_option_name)
+
+    log.debug("Current compression is %s", original_compression)
+    values = ['dc', 'all', 'none']
+    values_to_toggle = list(filter(lambda value: value != original_compression, values))
+    new_compression = random.choice(values_to_toggle)
+
+    with scylla_yaml() as props:
+        props[internode_compression_option_name] = new_compression
+
+    db_cluster.restart_scylla()
+
```

Comment:
Each test must check something.
For now, it doesn't check anything. Here is no validation of the config change.

So, if config change doesn't get applied, this test will still pass.

I consider it a bug.